circuit_puzzles/programs/surplus_start_auction.clsp

```diff
@@ -123,54 +129,61 @@
-                    ; make sure that start_time hasn't happen yet, allow it to be in mempool for 5 tx blocks
-                    (c (list ASSERT_BEFORE_SECONDS_ABSOLUTE (+ start_time (* 5 MAX_TX_BLOCK_TIME)))
-                      ; assert p2_surplus_auction coin received the lot amount
-                      (c (list RECEIVE_MESSAGE 0x3f (concat p2_surplus_auction_coin_id lot_amount ()) p2_surplus_auction_coin_id)
-                        (c (list REMARK my_coin_id lot_amount 0 () CAT_TAIL_HASH CRT_TAIL_HASH)
-                          (assert-treasury-coins-withdrawal
-                            p2_treasury_hash
-                            CAT_MOD_HASH
-                            CAT_TAIL_HASH
-                            STATUTES_STRUCT
-                            treasury_maximum
-                            statutes_puzzle_hash
-                            treasury_coins
-                            my_coin_id
-                            lot_amount ; lot amount of byc offered
-                            0 ; withdrawal sum amount
-                            0 ; leftover byc amount
-                            announcements
-                          )
-                        )
-                      )
-                    )
-                  )
-                )
-              )
-            )
+        ; TODO: comment below can be deleted. requiring amount = 0 is not strictly speaking required.
+        ;   it simply stops us from launching multiple Surplus Auctions from same parent coin.
+        ;; it's important to force the amount to zero, since we don't want parent to create multiple coins
+        ;; with the same amount and launcher_id, which would enable sibling coins to mess with the state and takeover
+        ;; the auction and the lot
+        (list ASSERT_MY_AMOUNT 0)
+        (list ASSERT_MY_COIN_ID my_coin_id)
+        (assert-statute statutes_puzzle_hash STATUTE_SURPLUS_AUCTION_LOT lot_amount)
+        (assert-statute statutes_puzzle_hash STATUTE_P2_TREASURY_MOD_HASH p2_treasury_hash)
+        (assert-statute statutes_puzzle_hash STATUTE_P2_SURPLUS_AUCTION_MOD_HASH p2_surplus_auction_mod_hash)
+        ; start_time minus one tx block time should already be in the past
+        (list ASSERT_SECONDS_ABSOLUTE (- start_time MAX_TX_BLOCK_TIME))
+        ; make sure that start_time hasn't happen yet, allow it to be in mempool for 5 tx blocks
+        (list ASSERT_BEFORE_SECONDS_ABSOLUTE (+ start_time (* 5 MAX_TX_BLOCK_TIME)))
+        ; assert p2_surplus_auction coin received the lot amount
+        (list
+          RECEIVE_MESSAGE
+          0x3f
+          (concat
+            PROTOCOL_PREFIX
+            ;p2_surplus_auction_coin_id
+            lot_amount
+            ()
           )
+          p2_surplus_auction_coin_id
+        )
+        (list
+          REMARK
+          PROTOCOL_PREFIX
+          my_coin_id ; TODO: delete as provides no new information to driver
```

Comment:
do you want to delete this?

Reply:
let's actually keep it to match remark format of bid operation. we also don't lose anything because my_coin_id needs to get passed in via solution regardless